### PR TITLE
Add progress bar

### PR DESCRIPTION
**Pull request by arianna-h-h**

### CODE REVIEW

File: gistUpload.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 const axios = require('axios');
-const fs = require('fs');
+const fileSystem = require('fs');
 
 const API_URL = 'https://api.github.com/gists';
+let finalString = '';
```
Comment (snmgian) [inline gistUpload.js:5]: What will occur if you call `readFile('someFile', 'someFile')`, and then `readFile('someOtherFile', 'someOtherFile')` ?

File: gistUpload.js [hunk 2]
```diff
@@ -14,17 +15,40 @@ function uploadFile(contents, gist) {
...
+  readStream.on('data', (chunk) => {
+    try {
+      addToFile(chunk);
+      bytesRead = chunk.length + bytesRead;
+      console.log(`${chunk.length} bytes in this chunk.\n`);
+      const prog = (Math.round((readStream.bytesRead / totalSize) * 100) - 15);
+      console.log(`${prog}% of file has been read`);
+      console.log(`${bytesRead} bytes read so far\n`);
+    } catch (error) {
+      console.log(error);
```
Comment (snmgian) [inline gistUpload.js]: If an error occurs we don't want to upload the file, right ? How are we preventing the upload in that case?

File: gistUpload.js [hunk 3]
```diff
@@ -14,17 +15,40 @@ function uploadFile(contents, gist) {
...
+  });
+
+  readStream.on('close', () => {
+    uploadFile(finalString, gistToName);
+  });
 }
 
 
-module.exports = { uploadFile, readFile };
+module.exports = { readFile, uploadFile };
```
Comment (snmgian) [inline gistUpload.js:52]: Do you need to export `uploadFile` ?

File: gistUpload.js [hunk 4]
```diff
@@ -14,17 +15,40 @@ function uploadFile(contents, gist) {
...
+      const prog = (Math.round((readStream.bytesRead / totalSize) * 100) - 15);
+      console.log(`${prog}% of file has been read`);
+      console.log(`${bytesRead} bytes read so far\n`);
+    } catch (error) {
+      console.log(error);
+    }
+  });
+
+  readStream.on('close', () => {
+    uploadFile(finalString, gistToName);
```
Comment (snmgian) [inline gistUpload.js:47]: See this: https://github.com/jasonsutter87/Gist-uploader/pull/7#discussion_r138871711

I'm not requesting to follow promises / async approach in this PR, just add it to your todo's list

**Review by snmgian — CHANGES_REQUESTED** [review]

File: gistUpload.js (hunk 4 above)
Comment (arianna-h-h) [reply gistUpload.js:47]: done

File: gistUpload.js (hunk 3 above)
Comment (arianna-h-h) [reply gistUpload.js:52]: Yes, because I call it in my gistUpload.test.js

File: gistUpload.js (hunk 1 above)
Comment (arianna-h-h) [reply gistUpload.js:5]: The same file gets uploaded twice, once with the different gist names.

File: gistUpload.js (hunk 2 above)
Comment (arianna-h-h) [reply gistUpload.js]: done

File: gistUpload.js [hunk 5]
```diff
@@ -14,17 +15,40 @@ function uploadFile(contents, gist) {
...
+  let bytesRead = 0;
+  const totalSize = fileSystem.statSync(process.argv[2]).size;
+  console.log(`Total size: ${totalSize}`);
+  const readStream = fileSystem.createReadStream(fileToContent);
+
+  readStream.on('data', (chunk) => {
+    try {
+      addToFile(chunk);
+      bytesRead = chunk.length + bytesRead;
+      console.log(`${chunk.length} bytes in this chunk.\n`);
```
Comment (ghaynes1292) [inline gistUpload.js]: look into `process.stdout.write():`

**Review by ghaynes1292 — CHANGES_REQUESTED** [review]

**Review by ghaynes1292 — APPROVED** [review]

**Pull request merged**